Sources/SourceKitLSP/Swift/DocumentFormatting.swift

```diff
@@ -139,26 +139,58 @@ private func edits(from original: DocumentSnapshot, to edited: String) -> [TextE
 extension SwiftLanguageService {
   package func documentFormatting(_ req: DocumentFormattingRequest) async throws -> [TextEdit]? {
     return try await format(
+      snapshot: documentManager.latestSnapshot(req.textDocument.uri),
       textDocument: req.textDocument,
       options: req.options
     )
   }
 
   package func documentRangeFormatting(_ req: DocumentRangeFormattingRequest) async throws -> [TextEdit]? {
     return try await format(
+      snapshot: documentManager.latestSnapshot(req.textDocument.uri),
       textDocument: req.textDocument,
       options: req.options,
       range: req.range
     )
   }
 
+  package func documentOnTypeFormatting(_ req: DocumentOnTypeFormattingRequest) async throws -> [TextEdit]? {
+    guard let server = self.sourceKitLSPServer else {
+      return nil
+    }
+    let snapshot = try documentManager.latestSnapshot(req.textDocument.uri)
+    let capabilities = await server.serverCapabilities(
+      for: capabilityRegistry.clientCapabilities,
+      registry: capabilityRegistry
+    )
+    
+    guard let documentOnTypeFormattingProvider = capabilities.documentOnTypeFormattingProvider,
+      documentOnTypeFormattingProvider.firstTriggerCharacter == req.ch
+        || documentOnTypeFormattingProvider.moreTriggerCharacter?.contains(req.ch) == true,
+      let line = snapshot.lineTable.line(at: req.position.line),
+      /// No need to go through whitespace checking if the trigger is not a newline
```

Comment:
This isn’t a doc comment, so it should  have only two slashes.

Reply:
I'm used to use doc comments everywhere since it comes with some markdown capabilities.
But I understand you're following the more "standardized" way of writing comments in this project.